module/platform/darwin/SensorFilter/src/SensorFilter.cpp

```diff
@@ -491,43 +514,37 @@ namespace module::platform::darwin {
                                              config.motionFilter.noise.measurement.flatFootOrientation,
                                              MeasurementType::FLAT_FOOT_ORIENTATION());
                     }
+                    // Else the foot is off the ground, so we make sure that for the next time step, we know that
+                    // this time step, the foot was off the ground
                     else if (!foot_down) {
                         previous_foot_down[side] = false;
                     }
 
+                    // Note that the Hwf is set, even if the foot is not down. This means that moving feet in the air
+                    // will have an Hwf associated with them which is the transform from when that foot last hit
+                    // the ground
                     sensors->feet[side].Hwf = footlanding_Hwf[side].matrix();
                 }
 
                 // Calculate our time offset from the last read
-                double deltaT = std::max(
+                // too hard to read
+                const double deltaT = std::max(
                     (input.timestamp - (previousSensors ? previousSensors->timestamp : input.timestamp)).count()
                         / double(NUClear::clock::period::den),
                     0.0);
```

Comment:
this is unreadable IMO

Reply:
Yeah and it's doing dumb stuff... Probably written by me before I knew the std::chrono library better